rmw_implementation/src/functions.cpp

```diff
@@ -56,27 +92,40 @@ load_library()
     return nullptr;
   }
 
-  if (env_var.empty()) {
-    env_var = STRINGIFY(DEFAULT_RMW_IMPLEMENTATION);
+  // User specified an RMW, attempt to load that one and only that one
+  if (!env_var.empty()) {
+    return attempt_to_load_one_rmw(env_var);
   }
 
-  std::string library_name;
-  try {
-    library_name = rcpputils::get_platform_library_name(env_var);
-  } catch (const std::exception & e) {
-    RMW_SET_ERROR_MSG_WITH_FORMAT_STRING(
-      "failed to compute shared library name due to %s", e.what());
-    return nullptr;
+  // User didn't specify, so next try to load the default RMW
+  rmw_reset_error();
```

Comment:
nitpick: why does it need to call `rmw_reset_error` here? it cannot get to here if any failure happens before in this function. so i was expecting no error would be set here.

Reply:
Oh, good point.  That can never happen here.  I'll remove this line, thanks for the review.